scripts/generate-types.js

```diff
@@ -0,0 +1,65 @@
+const fs = require('fs')
+
+const pascalCase = require('pascal-case')
+const prettier = require('prettier')
+const TypeWriter = require('@gimenete/type-writer')
+const webhooks = require('@octokit/webhooks-definitions')
+
+const signatures = []
+const tw = new TypeWriter()
+
+webhooks.forEach(({ name, actions, examples }) => {
+  if (!examples) {
+    return
+  }
+
+  const typeName = `WebhookPayload${pascalCase(name)}`
+  tw.add(examples, {
+    rootTypeName: typeName,
+    namedKeyPaths: { [`${typeName}.repository`]: 'PayloadRepository' }
+  })
+
+  const events = [
+    `'${name}'`,
+    ...actions.map(action => `'${name}.${action}'`)
+  ].join(' | ')
+  signatures.push(`
+    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<${typeName}>) => void): void
+  `)
+})
+
+const definition = `
+// DO NOT EDIT THIS FILE DIRECTLY
+// make edits in scripts/generate-types.js
+
+export type Options = {
+  path: string,
+  secret: string
+}
+
+${tw.generate('typescript', { inlined: false }).replace(/type /g, 'export type ')}
+
+export namespace Webhooks {
+  export interface WebhookEvent<T> {
+    id: string
+    name: string
+    payload: T
+    protocol?: 'http' | 'https'
+    host?: string
+    url?: string
```

Comment:
@gimenete where does `protocol`, `host`, `url` come from?

Reply:
I don’t know. I copied that from the definitions probot had